homework/index.js

```diff
@@ -30,18 +17,144 @@
+    setInnerText(5, upDate);
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  function renderContributors(contributorsContainer, contributors) {
+    contributors.forEach(contributor => {
+      const ul = createAndAppend('ul', contributorsContainer);
+      const li = createAndAppend('li', ul);
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        class: 'image',
+        height: '48',
+      });
+      const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contributorData, {
+        text: contributor.login,
+        class: 'contributor-name',
+      });
+      createAndAppend('div', contributorData, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.addEventListener('click', () => {
+        window.open(contributor.html_url, '_blank');
+      });
+    });
+  }
+
+  function presentContributors(contributorsContainer, url) {
+    fetch(url)
+      .then(contributors => contributors.json())
+      .then(contributors => {
+        createAndAppend('p', contributorsContainer, { text: 'Contributions' });
+        renderContributors(contributorsContainer, contributors);
+      })
+      .catch(showError);
+  }
 
+  function sortRepositories(repositories) {
+    repositories.sort((a, b) => a.name.localeCompare(b.name, 'en', { sensitivity: 'base' }));
+  }
+
+  function selectStructure(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        value: index,
+        id: repository.name,
+        text: repository.name,
+      });
+    });
+  }
+
+  function showAll(repositories) {
+    const header = createAndAppend('div', root, { id: 'header' });
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header, { id: 'select' });
+    const mainContainer = createAndAppend('div', root, { id: 'main-container' });
+    const repositoriesContainer = createAndAppend('div', mainContainer, {
+      id: 'repositories-container',
+    });
+    const contributorsContainer = createAndAppend('div', mainContainer, {
+      id: 'contributors-container',
+    });
+
+    sortRepositories(repositories);
+    selectStructure(repositories, select);
+
+    presentRepositories(repositoriesContainer, repositories[0]);
+    presentContributors(contributorsContainer, repositories[0].contributors_url);
+
+    select.addEventListener('change', () => {
+      repositoriesContainer.innerHTML = '';
+      contributorsContainer.innerHTML = '';
+      const index = select.selectedIndex;
+      presentRepositories(repositoriesContainer, repositories[index]);
+      presentContributors(contributorsContainer, repositories[index].contributors_url);
+    });
+  }
+
+  function main(url) {
+    fetch(url)
+      .then(response => {
+        if (response.ok) {
```

Comment:
Hi Ammar - for this assignment, you need not have this check for `response.ok`. In case there is any issue in fetching response from GitHub, it is automatically going to go inside the catch block. 🙂

Reply:
Hi Ammor - my apologies. I forgot that `fetch` doesn't rejects a promise in case there is an error while fetching a resource. In that case, the check that you have added is necessary. Please ignore my above comment. 🙂 

Thank you @remarcmij for correcting me here.

Reply:
:pray: :slightly_smiling_face: